pkg/controller/egress/controller.go

```diff
@@ -466,6 +517,31 @@ func (c *EgressController) syncEgress(key string) error {
 	return nil
 }
 
+func (c *EgressController) externalIPPoolWorker() {
+	for c.processNextExternalIPPoolWorkItem() {
+	}
+}
+
+func (c *EgressController) processNextExternalIPPoolWorkItem() bool {
+	key, quit := c.poolQueue.Get()
+	if quit {
+		return false
+	}
+	defer c.poolQueue.Done(key)
+
+	err := c.updateExternalIPPoolStatus(key.(string))
+	if err != nil {
+		// Put the item back on the workqueue to handle any transient errors.
```

Comment:
nit: s/on the workqueue/in the workqueue

Reply:
Done. I saw both 'on the workqueue' and 'in the workqueue' in our project. 
Do we need to unify this? If yes I think I can implement in this PR.
Some projects use "on the workqueue":
https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/cloud-provider/controllers/node/node_controller.go

Reply:
it's possible that both are correct or maybe I'm wrong :). either way both versions are easy to understand.